docs: update documentation for the version of the builders (#155)
aica-technology/api: docs/docs/reference/custom-components/05-aica-package-toml.md

the Docker syntax and version used to build the package.

```toml title="aica-package.toml"
#syntax=ghcr.io/aica-technology/package-builder:v1
```

You can find the available
versions [here](https://github.com/orgs/aica-technology/packages/container/package/package-builder).

> The upcoming sections describe the syntax for the `package-builder` version `1.0.0` and later. If you are migrating from an older version, please refer to any of the following guide:

Reviewer: nitpick
```suggestion
The upcoming sections describe the syntax for the `package-builder` version `1.0.0` and later. If you are migrating from an older version, please refer to the corresponding guide:
```

#### `[build.ssh]`

Optional. Default is `false`. If set to `true`, any call to `CMake`, `pip` or custom stage will have access to the SSH credentials given to Docker with `--ssh default`. This is useful if you need to clone a private repository in CMakeLists.txt or install a private Python package from git.

```toml title="aica-package.toml"
[build]
ssh = false
```

#### `[build.image]`

Required. `[build.image]` is the tag of the AICA `ghcr.io/aica-technology/ros2-ws` image that will be used to build the components. Those images are tagged after the versions of the ROS 2 distributions and are available [here](https://github.com/aica-technology/docker-images/pkgs/container/ros2-ws).

```toml title="aica-package.toml"
[build]
image = "iron"
```

#### `[build.cmake-args]`

Optional. In this category, you can specify arguments which will be passed to CMake through `-DK1=V1 -DK2=V2 ...`

```toml title="aica-package.toml"
[build.cmake-args]
# here are a few examples
PACKAGE_NAME = "my_package"
USE_FEATURE_X = "ON"
```

#### `[build.dependencies]`

Optional. `[build.dependencies]` is used to specify the AICA libraries and ROS 2 packages that will be installed in the
image.

:::note

> Those libraries and packages will not be available at runtime, only while building. If you need them for building _and_ running, add them to the `[build.packages.XYZ.dependencies]` section below instead.

Reviewer: Might be nice to add a note here along the lines of "Libraries that are only required at runtime, and not at build time, can be declared with `build.run-dependencies` instead"

"""
```

### `[metadata]`

Required.

#### `[metadata.version]`

Required. Must be a [semver-compliant](https://semver.org/) version string.

```toml title="aica-package.toml"
[metadata]
version = "1.0.0"
```

#### `[metadata.description]`

Optional. A short description of the component.

```toml title="aica-package.toml"
[metadata]
description = "My awesome component"
```

#### `[metadata.collection]`

Required when using multiple `[build.packages]`, otherwise ignored. This must contain the names of the collection being built (the set of multiple components).

`[metadata.collection.name]` is required. It is the name of the collection.

`[metadata.collection.ros-name]` is optional. This must be specified if `[metadata.collection.name]` is not a valid ROS package name.

```toml title="aica-package.toml"
[metadata.collection]
name = "my-collection"
ros-name = "my_collection" # required because `my-collection` is not a valid ROS package name
```

#### Advanced usage

> ##### `docker-image://`

Reviewer: In this section and in build-context, you still refer to `[build.environment]` (`[build.environment.aica]` / `[build.environment.aica.libraries]` / `[build.environment.aica.ros]`)

source = "build-context://my_source"
```

##### Version constraints

Version constraints follow the syntax of the [Terraform version constraints](https://www.terraform.io/docs/language/expressions/version-constraints.html) which is similar to syntax used by NPM, yarn or pip. Here is a quick summary:

- Versions constraints are composed of one or more conditions separated by commas, e.g. `>= 1.0.0, < 2.0.0`.
- Each version specified must be a valid [semver](https://semver.org/) version, e.g. `1.0.0`.
- The following operators are supported:
  - `=` or no operator: allow only the exact version, cannot be combined with other conditions.
  - `!=`: exclude a specific version.
  - `>`, `>=`, `<`, `<=`: comparison against a specific version, allowing any version matching the operator. `>` allows newer version and `<` allows older version.
  - `~>`: allows only the right most number of the version to increase. This is useful to allow only patch or minor versions to increase, e.g. `~> 1.0` allows `1.1`, `1.2`, etc. but not `2.0` and `~> 1.0.0` allows `1.0.1`, `1.0.2`, etc. but not `1.1.0`.

### Examples

> #### Basic setup

Reviewer: The examples are still on `#syntax=ghcr.io/aica-technology/package-builder:v0.0.13`, wouldn't you prefer to update those too?

Author: This is crazy, I went through every page multiple times, did search and everything and somehow I still missed 4 of them? 😂
